pylint/testutils/_primer/primer_run_command.py

```diff
@@ -30,14 +30,22 @@ class RunCommand(PrimerCommand):
     def run(self) -> None:
         packages: PackageMessages = {}
         fatal_msgs: list[Message] = []
-        for package, data in self.packages.items():
+        package_data_iter = (
+            self.packages.items()
+            if self.config.batches is None
+            else list(self.packages.items())[
+                self.config.batchIdx :: self.config.batches
+            ]
+        )
+        for package, data in package_data_iter:
             messages, p_fatal_msgs = self._lint_package(package, data)
             fatal_msgs += p_fatal_msgs
             local_commit = Repo(data.clone_directory).head.object.hexsha
             packages[package] = PackageData(commit=local_commit, messages=messages)
         path = (
             self.primer_directory
-            / f"output_{'.'.join(str(i) for i in sys.version_info[:3])}_{self.config.type}.txt"
+            / f"output_{'.'.join(str(i) for i in sys.version_info[:3])}_{self.config.type}"
+            f"_batch{self.config.batchIdx}.txt"
```

Comment:
Should we add a check here whether there are batches defined? Otherwise this name doesn't make a lot of sense.

Reply:
Right, it will just say `_batchNone` as of now. I can clean this up a little.

Reply:
b94d0131